service/storage_proxy.cc

```diff
@@ -539,13 +559,14 @@ class storage_proxy::remote {
         auto schema_version = decision.update.schema_version();
         return handle_write(src_addr, t, schema_version, std::move(decision), std::move(forward), reply_to, shard,
                 response_id, trace_info,
+                fencing_token{},
```

Comment:
Why don't we pass it for paxos?

Reply:
skipped paxos RPCs in this PR, to restrict the scope of the PR/get early feedback

Reply:
> v2:
> 
> * addressed all the comments

That's really making me re-review everything. With a real changelog I can focus on the more important changes.


> * converted the spec to .md
> * added description to the PR
> 
> @avikivity @gleb-cloudius please re-review

Reply:
Don't know why this got threaded here.

Reply:
> That's really making me re-review everything. With a real changelog I can focus on the more important changes.

* `fence_version` moved from `topology` table to `scylla_local`;
* `docs/dev/raft-topology.md` is added;
* changed version in RPC description to `[[version 5.4.0]]`;
* removed `param` from raft commands, use current version after `read_barrier`;
* added `cmd_index` to raft commands to protect agains commands reordering;
* `try_apply_fence` renamed to `apply_fence`, added overload with `future` parameter;
* specific type instead of `std::monostate()` in `register_hint_mutation`;
* use `apply_fence` with future overload to fence read and write requests;
* added fencing to `send_to_live_endpoints`, handle `stale_topology_exception`;

Reply:
Thanks.

Loading from scylla_local makes sense (and answers my question about it being raft controlled), but we should make an effort to sync with the cluster before accepting ops, to avoid needless fencing.